store/tikv/region_cache.go

```diff
@@ -1320,8 +1320,18 @@ func (r *Region) FollowerStorePeer(rs *RegionStore, followerStoreSeed uint32, op
 }
 
 // AnyStorePeer returns a leader or follower store with the associated peer.
+// Not sent to the learner peer.
+// TODO: better to retry by tikv response's error message.
 func (r *Region) AnyStorePeer(rs *RegionStore, followerStoreSeed uint32, op *storeSelectorOp) (store *Store, peer *metapb.Peer, accessIdx AccessIndex, storeIdx int) {
-	return r.getKvStorePeer(rs, rs.kvPeer(followerStoreSeed, op))
+	for i := 0; i < rs.accessStoreNum(TiKVOnly); i++ {
+		store, peer, accessIdx, storeIdx = r.getKvStorePeer(rs, rs.kvPeer(followerStoreSeed, op))
+		if peer.Role == metapb.PeerRole_Learner {
+			followerStoreSeed++
+			continue
+		}
+		return
+	}
+	return
```

Comment:
Maybe doing it in `kvPeer()` will be clearer? `kvPeer()` is only called by `AnyStorePeer`.